network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/exclusion/ui/AppExclusionListAdapter.kt

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright (c) 2023 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.networkprotection.impl.exclusion.ui
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.RecyclerView
+import com.duckduckgo.app.global.extensions.safeGetApplicationIcon
+import com.duckduckgo.mobile.android.ui.view.gone
+import com.duckduckgo.mobile.android.ui.view.show
+import com.duckduckgo.networkprotection.impl.R
+import com.duckduckgo.networkprotection.impl.databinding.ItemExclusionListAppBinding
+import com.duckduckgo.networkprotection.impl.databinding.ItemExclusionListFilterBinding
+import com.duckduckgo.networkprotection.impl.databinding.ItemExclusionListHeaderBinding
+import com.duckduckgo.networkprotection.impl.exclusion.ui.AppsProtectionType.AppType
+import com.duckduckgo.networkprotection.impl.exclusion.ui.AppsProtectionType.FilterType
+import com.duckduckgo.networkprotection.impl.exclusion.ui.AppsProtectionType.HeaderType
+
+class AppExclusionListAdapter(
+    val listener: ExclusionListListener,
+) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
```

Comment:
ListAdapter<AppsProtectionType, RecyclerView.ViewHolder>(SimpleDiffUtil())

where SimpleDiffUtil is

```kotlin
@Suppress("FunctionName")
inline fun <reified R : Any> SimpleDiffUtil() = object : DiffUtil.ItemCallback<R>() {
    override fun areItemsTheSame(oldItem: R, newItem: R) = oldItem::class == newItem::class

    @SuppressLint("DiffUtilEquals")
    override fun areContentsTheSame(oldItem: R, newItem: R) = oldItem == newItem
}
```

Could remove huge part of code from this adapter.
Check also `com.duckduckgo.mobile.android.themepreview.ui.component.ComponentAdapter` for reference in your codebase.